library/FrontRow/App/Database.hs

```diff
@@ -58,7 +62,13 @@ runDB
   -> m a
 runDB action = do
   pool <- asks getSqlPool
-  runSqlPool action pool
+  mTimeoutSeconds <- asks getStatementTimeoutSeconds
+
+  flip runSqlPool pool $ do
+    for_ mTimeoutSeconds $ \timeoutSeconds ->
+      let timeoutMilliseconds = timeoutSeconds * 1000
+      in [executeQQ| SET statement_timeout = #{timeoutMilliseconds} |]
+    action
 
 data PostgresConnectionConf = PostgresConnectionConf
```

Comment:
Should we extend this (and its `env` parser) to look for (something like) `PGSTATEMENT_TIMEOUT`?

I guess it might not be as valuable, since the `PostgresConnectionConf` is typically read, used to make the `Pool SqlBackend` then discarded. Our users would need to stash just part of it on their `App` to use in the instance later. Might still be worth it as a way to standardize the env var.

Reply:
I take back my hesitance here. I think the common pattern is,

```hs
data AppSettings =
  appPostgresConf :: PostgresConnectionConf

data App =
  appSettings :: AppSettings
```

So this would be useful,

```hs
instance HasSqlPool App where
  -- ...

  getStatementTimeout = pccStatemetnTimeout . appPostgresConf . appSettings
```

Reply:
A bit confused. The `PGConf` is only used in `makePostgresPoolWith`, and everything downstream relies on the result type of `SqlPool`. The only `instance` declaration we've got is:
```haskell
instance HasSqlPool SqlPool where
  getSqlPool = id
  getStatementTimeout = const Nothing
```
which has no easy way to anticipate the presence of that conf object?

Reply:
Right, that was my initial hesitance

> I guess it might not be as valuable, since the PostgresConnectionConf is typically read, used to make the Pool SqlBackend then discarded

But then I realized the `AppSettings` pattern, which is common.

> The only instance declaration we've got is:

If you look in `jobs`, you'll see the non-`id` usage:

```hs
data App = App
  { appQueue :: Queue
  , appLogLevel :: LogLevel
  -- ...
  , appSqlPool :: SqlPool
  , -- ...
  }

instance HasLogging App where
  getLogLevel = appLogLevel

instance HasSqlPool App where
  getSqlPool = appSqlPool

loadApp :: SqlPool -> IO App
loadApp pool = do
  (..., ..., ...) <- Env.parse $ (,,,,,,,,,,)
    <$> Env.var Env.auto "QUEUE" Env.nonEmpty
    <*> Env.flag (Env.Off LevelInfo) (Env.On LevelDebug) "DEBUG" mempty

  -- ...

  pure $ App
    { appQueue = queue
    , appLogLevel = logLevel
    , appLogFormat = logFormat
    , appSqlPool = pool
    }
```

This is actually a **bad** example:

- It's taking the `pool` into `loadApp` instead of making it there; a conventional `loadApp` should be `:: IO App`
- It's parsing ENV into a massive tuple, ick

This should really be:

```hs
data AppSettings = AppSettings
  { appQueue :: Queue
  , appLogLevel :: LogLevel
  , appPostgresConf :: PostgresConnectionConf -- NEW
  }

instance HasLogging AppSettings where
  getLogLevel = appLogLevel

-- Look ma, no tuple
loadAppSettings :: IO AppSettings
loadAppSettings = Env.parse $ AppSettings
    <$> Env.var Env.auto "QUEUE" Env.nonEmpty
    <*> Env.flag (Env.Off LevelInfo) (Env.On LevelDebug) "DEBUG" mempty
    <*> envParsePostgresConfg -- NEW

data App = App
  { appSettings :: AppSettings
  , appSqlPool :: SqlPool
  }

instance HasLogging App where
  getLogLevel = getLogLevel . appSettings

instance HasSqlPool where
  getSqlPool = appSqlPool
  getStatementTimeout = pccStatementTimeout . appPostgresConf . appSettings -- NEW

loadApp :: IO App
  appSettings@AppSettings {..} <- loadAppSettings
  appSqlPool <- makePostgresPoolWith appPostgresConf  
  pure App { .. }
```

Now, I have a feeling we didn't (or can't) do this because of some complexities with building the Postgres connection using the dynamic IAM credentials. If that's true, fine, we won't refactor Jobs like this. But I still think `PostgresConnectionConfig` should hold the timeout too, and the `env` parser should get it for you, since that makes sense (to me) from the library's perspective.

Not blocking the PR (never was), so feel free to punt on it.

Reply:
Gotcha gotcha. Definitely prefer the pattern of centralizing and preserving `AppSettings`. Happy to try to bend the universe towards that,